src/models/ProactiveReportingConfigs.ts

```diff
@@ -0,0 +1,71 @@
+export default class ProactiveReportingConfigs {
+  public readonly gapBetweenModals: number; // Time in seconds
+  public readonly modalDelayAfterDetection: number; // Time in seconds
+  public readonly enabled: boolean;
+
+  // Private constructor to ensure it can only be created through the builder
+  private constructor(
+    gapBetweenModals: number,
+    modalDelayAfterDetection: number,
+    enabled: boolean,
+  ) {
+    this.gapBetweenModals = gapBetweenModals;
+    this.modalDelayAfterDetection = modalDelayAfterDetection;
+    this.enabled = enabled;
+  }
+
+  // Static method to get the builder instance
+  static Builder = class {
+    public gapBetweenModals: number = 30; // Default: 30 seconds
+    public modalDelayAfterDetection: number = 15; // Default: 15 seconds
+    public enabled: boolean = true; // Default: enabled
+
+    // Logger method to handle logging
+    public logWarning(message: string): void {
+      console.warn(`Warning: ${message}`);
+    }
```

Comment:
I think we can use `Logger.warn` instead of this so it follows logging configuration set by user